pkg/plugin/aggregation/run_test.go

```diff
@@ -0,0 +1,198 @@
+package aggregation
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/driver"
+	"github.com/heptio/sonobuoy/pkg/plugin/driver/job"
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+)
+
+func TestRunAndMonitorPlugin(t *testing.T) {
+	// Dead simple plugin works for this test. No need to test daemonset/job specific logic so
+	// a job plugin is much simpler to test against.
+	testPlugin := &job.Plugin{
+		Base: driver.Base{
+			Definition: plugin.Definition{
+				Name:       "myPlugin",
+				ResultType: "myPlugin",
+			},
+			Namespace: "testNS",
+		},
+	}
+	testPluginExpectedResults := []plugin.ExpectedResult{
+		{ResultType: "myPlugin"},
+	}
+	healthyPod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+	}
+	failingPod := corev1.Pod{
```

Comment:
Idea for future: Now that these have been introduced in a number of tests, it would be nice to have some utilities to help with creating these healthy and failing pod objects.

Reply:
Yea, its pretty verbose and could be minimized a bit between the daemonset, job, and now aggregation tests.